plugins/identity-shiro/src/main/java/org/opensearch/identity/shiro/ShiroTokenHandler.java

```diff
@@ -0,0 +1,97 @@
+import org.apache.shiro.authc.AuthenticationToken;
+import org.apache.shiro.authc.UsernamePasswordToken;
+import org.opensearch.identity.Subject;
+import org.opensearch.identity.tokens.AuthToken;
+import org.opensearch.identity.tokens.BasicAuthToken;
+import org.opensearch.identity.tokens.TokenManager;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * Extracts Shiro's {@link AuthenticationToken} from different types of auth headers
+ *
+ * @opensearch.experimental
+ */
+class ShiroTokenHandler implements TokenManager {
+
+    /**
+     * Translates into shiro auth token from the given header token
+     * @param authenticationToken the token from which to translate
+     * @return An optional of the shiro auth token for login
+     */
+    public Optional<AuthenticationToken> translateAuthToken(org.opensearch.identity.tokens.AuthToken authenticationToken) {
+        if (authenticationToken instanceof BasicAuthToken) {
+            final BasicAuthToken basicAuthToken = (BasicAuthToken) authenticationToken;
+            return Optional.of(new UsernamePasswordToken(basicAuthToken.getUser(), basicAuthToken.getPassword()));
+        }
+
+        return Optional.empty();
+    }
+
+    @Override
+    public AuthToken issueToken() {
+
+        Subject subject = new ShiroSubject(this, SecurityUtils.getSubject());
+        final byte[] rawEncoded = Base64.getEncoder().encode((subject.getPrincipal().getName() + ":" + generatePassword()).getBytes(UTF_8));
+        final String usernamePassword = new String(rawEncoded, UTF_8);
+        final String header = "Basic " + usernamePassword;
+
+        return new BasicAuthToken(header);
+    }
+
+    @Override
+    public boolean validateToken(AuthToken token) {
+        if (token instanceof BasicAuthToken) {
+            final BasicAuthToken basicAuthToken = (BasicAuthToken) token;
+            if (basicAuthToken.getUser().equals(SecurityUtils.getSubject()) && basicAuthToken.getPassword().equals(generatePassword())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public String getTokenInfo(AuthToken token) {
+        if (token instanceof BasicAuthToken) {
+            final BasicAuthToken basicAuthToken = (BasicAuthToken) token;
+            return basicAuthToken.toString();
+        }
+        throw new UnsupportedAuthenticationToken();
+    }
+
+    @Override
+    public void revokeToken(AuthToken token) {
+        if (token instanceof BasicAuthToken) {
+            final BasicAuthToken basicAuthToken = (BasicAuthToken) token;
+            basicAuthToken.revoke();
+            return;
+        }
+        throw new UnsupportedAuthenticationToken();
+    }
+
+    @Override
+    public void resetToken(AuthToken token) {
+        if (token instanceof BasicAuthToken) {
+            final BasicAuthToken basicAuthToken = (BasicAuthToken) token;
+            basicAuthToken.revoke();
+        }
+    }
+
+    public String generatePassword() {
```

Comment:
Hardcoded?

Reply:
This is just for testing/demo. You would not actually use the Shiro Identity Plugin for anything, so it is just hardcoded. The Security Plugin will have a truly random password generated.

Reply:
It is a bit confusing to have this into the main codebase, why do we need this plugin if it is not used? if this is an example, we have dedicated place for such plugins, https://github.com/opensearch-project/OpenSearch/tree/main/plugins/examples

Reply:
Yeah, that is a fair comment. Basically, the Shiro plugin allows for running the core codebase with Identity enabled but the Security plugin not installed. You can think of it like a mini-security plugin for the time being. The reason I said it is just an example for now is that there is not all of the functionality in place that the Security plugin offers. For the time being, you would not really use it in place of the Security plugin. Eventually we will be able to reproduce all the Security functionality directly in core. 

For the time being, I am more then happy to swap this to a randomized password if you think that is preferable.

Reply:
Thanks @scrawfor99 , I think this plugin belongs to `examples` (at least for the time being), since it is not required and as you mentioned, it is not usable as-is. What do you think?

Reply:
Sounds good, I will move it over.

Reply:
Hi @reta, sorry perhaps I was not clear before. While the current Shiro Identity plugin does not fully replicate the features of the Security Plugin, it does allow someone to turn on authentication without installing the Security Plugin. With the changes introduced to core, moving the plugin to the example package may cause a bit of confusion as it looks like these are mostly strictly demo packages. 

There are four main cases to consider:
1. The identity feature flag is not enabled-->Nothing happens with Security etc.; you have to use the legacy security system 
2. You turn on the Identity feature flag and have installed the Security Plugin--> The security plugin handles many complex use cases relating to Identity features. Specifically, it deals with Extension and TokenManagement 
3. You turn on the Identity feature flag but do not install the Security plugin--> The Shiro Plugin internally handles barebones authentication for the cluster. It is able to grant user based tokens for access 
4. You turn on the Identity feature flag but do not install the Security plugin and disable Shiro--> The Noop Identity Plugin acts as a pass through so that things don't break but you also do not get any features 

Given this clarification, do you still prefer moving the package to the examples? I have started the process and it seems doable. That being said, I do wonder if it fits in the same category as the rest of the examples.

Let me know.

Reply:
Gotcha, thanks @scrawfor99 , could we make this plugin (ShiroPlugin) a bit more usable? For example, use secure settings to fetch username / password? So at least, if installed, it could have basic means to be configured.

Reply:
Hi @reta, I think you mean @scrawfor99 not Craig haha. But yes, I agree. It is certainly a lot to juggle. I will take a look at the link you shared.

Reply:
Oh super apologies, thank you for correcting me  :pray:

Reply:
This was resolved in 1e9afe315b390b21fb5e31bbaea40a39b6da401d plugins/identity-shiro/src/main/java/org/opensearch/identity/shiro/ShiroTokenManager.java by switch to a randomly generated password